admin/tests/unit/controllers/display-languages.js

```diff
@@ -116,4 +118,132 @@ describe('Display Languages controller', function() {
       done();
     });
   });
+
+  it('should disable a language', (done) => {
+    settings.resolves({
+      locale: 'en',
+      locale_outgoing: 'sw',
+      languages: [
+        { locale: 'en', enabled: true },
+        { locale: 'sw', enabled: true },
+      ],
+    });
+    db.query.withArgs('medic-client/doc_by_type').resolves({
+      rows: [
+        {
+          id: 'messages-en',
+          doc: {
+            _id: 'messages-en',
+            code: 'en',
+            type: 'translations',
+            enabled: true,
+            generic: { 'a': 'a v1', 'b': 'b v1', 'c': 'c v1' },
+            custom: { 'a': 'a v2', 'c': 'c v2', 'd': 'd v1' }
+          }
+        },
+        {
+          id: 'messages-sw',
+          doc: {
+            _id: 'messages-sw',
+            code: 'sw',
+            type: 'translations',
+            enabled: true,
+            generic: {},
+            custom: { 'a': 'a v1', 'c': 'c v1', 'b': 'b v1', 'e': 'e v1'}
+          }
+        },
+      ]
+    });
+
+    createController();
+    setTimeout(() => {
+      rootScope.$digest();
+      const englishLanguage = scope.languagesModel.locales.find(locale => locale.doc.code === 'en');
+      chai.expect(englishLanguage.doc.code).to.equal('en');
+      chai.expect(englishLanguage.enabled).to.eq(true);
+      scope.disableLanguage(englishLanguage.doc);
```

Comment:
This test would be a lot cleaner if you changed `disableLanguage` to return the promise. That way instead of setting a timeout and hoping everything happens in the next tick you could just await the response.

Reply:
That's much cleaner indeed, thanks!